Joystream-Role-Bot/src/controls/control.ts

```diff
@@ -132,32 +154,65 @@ export const setMemberRole = async (client: Client): Promise<void> => {
     //   console.error("Error removing member roles:", error);
     // }
 
-    await discordMember.roles.add(RoleAddress.membershipLinked);
 
-    const roleUpdatePromises = qnMember.roles.map(async (memberRole) => {
-      const memberRoleGroupId = memberRole.groupId;
-      const mappedRoles = roleMap[memberRoleGroupId];
-      if (!mappedRoles) return;
+    const getRoles = discordMember.roles.cache.map((role: any) => role.id)
 
-      const [leadRoleId, workerRoleId] = mappedRoles;
-      const roleId = memberRole.isLead ? leadRoleId : workerRoleId;
-      const role = await guild.roles.fetch(roleId);
+    if (!getRoles.find((id: string) => id === RoleAddress.membershipLinked)) {
+      await discordMember.roles.add(linkRole);
+    }
 
-      if (!role) {
-        console.log(`<@&${roleId}> Role not found`);
-        return;
-      }
+    let nowWorking: GetWorkingState[] = [];
+
+    const roleUpdatePromises = qnMember.roles.map(async (memberRole: any) => {
+
+      const index = nowWorking.findIndex(data => data.workingGroups === memberRole.groupId)
 
-      const roleUpdatePromise =
-        memberRole.status.__typename === "WorkerStatusActive"
-          ? discordMember.roles.add(role)
-          : discordMember.roles.remove(role);
-      await roleUpdatePromise;
+      const active = memberRole.status.__typename === "WorkerStatusActive";
+      const lead = memberRole.isLead;
+      const workingGroups = memberRole.groupId;
+
+      if (index !== -1) {
+        nowWorking[index].active = active;
+        nowWorking[index].lead = lead;
+        nowWorking[index].workingGroups = workingGroups;
+      } else {
+        nowWorking.push({ active, lead, workingGroups });
+      }
     });
 
+    let worker = false;
+    if (nowWorking.length !== 0) {
+      nowWorking.map(async (work) => {
```

Comment:
Since we're not using return values in any way, let's replace `map` with `forEach`